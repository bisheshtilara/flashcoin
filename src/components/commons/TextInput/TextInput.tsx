import React, { InputHTMLAttributes } from "react";

export interface ITextInputProps extends InputHTMLAttributes<HTMLInputElement> {
  containerStyle?: string;
  inputStyle?: string;
  labelStyle?: string;
  label: string;
  icon?: JSX.Element;
}

const TextInput: React.FC<ITextInputProps> = ({
  containerStyle,
  inputStyle,
  labelStyle,
  label,
  type,
  value,
  icon,
  onChange,
}) => {
  return (
    <div className={`relative w-full ${containerStyle} rounded-lg`}>
      <p
        className={`absolute -top-3 left-4 bg-white px-2 text-base text-gray-400 font-extralight ${labelStyle}`}
      >
        {label}
      </p>
      <div>
        <input
          type={type}
          className={`border p-3 rounded-lg w-full focus:outline-red-600 ${inputStyle}`}
          value={value}
          onChange={onChange}
        />
        {icon && <div className="absolute top-3 right-3">{icon}</div>}
      </div>
    </div>
  );
};

export default TextInput;
