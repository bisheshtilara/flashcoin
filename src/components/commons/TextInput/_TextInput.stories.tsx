import { MagnifyingGlassIcon } from "@heroicons/react/24/outline";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import TextInput, { ITextInputProps } from "./TextInput";

export default {
  component: TextInput,
} as ComponentMeta<typeof TextInput>;


const defaultArgs: ITextInputProps = {
  label: "Label",
  type: "text",
  value: "",
};

const Template: ComponentStory<typeof TextInput> = (args: ITextInputProps) => (
  <TextInput {...args} />
);

export const Default = Template.bind({});
Default.args = {
  ...defaultArgs,
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  ...defaultArgs,
  icon: <MagnifyingGlassIcon className="h-6 text-red-500" />,
};

