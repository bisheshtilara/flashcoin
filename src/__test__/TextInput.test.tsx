import { MagnifyingGlassIcon } from "@heroicons/react/24/outline";
import { render, fireEvent, screen } from "@testing-library/react";
import TextInput from "../components/commons/TextInput";

describe("TextInput", () => {
    it("should render correctly", () => {
        const { container } = render(<TextInput label="test"/>);
        expect(container).toMatchSnapshot();
    });
    
    it("should render a input have icon", () => {
        const { container } = render(<TextInput label="test" icon={<MagnifyingGlassIcon className="h-6 text-red-600" />} />);
        expect(container).toMatchSnapshot();
    });
});