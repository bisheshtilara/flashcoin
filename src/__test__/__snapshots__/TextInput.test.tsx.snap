// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render a input have icon 1`] = `
<div>
  <div
    class="relative w-full undefined rounded-lg"
  >
    <p
      class="absolute -top-3 left-4 bg-white px-2 text-base text-gray-400 font-extralight undefined"
    >
      test
    </p>
    <div>
      <input
        class="border p-3 rounded-lg w-full focus:outline-red-600 undefined"
        value=""
      />
      <div
        class="absolute top-3 right-3"
      >
        <svg
          aria-hidden="true"
          class="h-6 text-red-600"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextInput should render correctly 1`] = `
<div>
  <div
    class="relative w-full undefined rounded-lg"
  >
    <p
      class="absolute -top-3 left-4 bg-white px-2 text-base text-gray-400 font-extralight undefined"
    >
      test
    </p>
    <div>
      <input
        class="border p-3 rounded-lg w-full focus:outline-red-600 undefined"
        value=""
      />
    </div>
  </div>
</div>
`;
